durabletask/worker.py

```diff
@@ -88,15 +90,25 @@ def __init__(self, *,
                  metadata: Optional[list[tuple[str, str]]] = None,
                  log_handler=None,
                  log_formatter: Optional[logging.Formatter] = None,
-                 secure_channel: bool = False):
+                 secure_channel: bool = False,
+                 interceptors: Optional[list[grpc.ServerInterceptor]] = None):  # Add interceptors
         self._registry = _Registry()
         self._host_address = host_address if host_address else shared.get_default_host_address()
-        self._metadata = metadata
+        self._metadata = metadata or []  # Ensure metadata is never None
```

Comment:
I don't think we have this same check in client.py. Should we? Or do we even need this?